code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -80,16 +80,82 @@
 	desc = "A sturdy shotgun with a longer magazine and a fixed tactical stock designed for non-lethal riot control."
 	icon_state = "riotshotgun"
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/riot
-	sawn_desc = "Come with me if you want to live."
+	var/style = 1
 
 /obj/item/weapon/gun/projectile/shotgun/riot/attackby(obj/item/A, mob/user, params)
 	..()
+	var/temp_sawing = 0
 	if(istype(A, /obj/item/weapon/circular_saw) || istype(A, /obj/item/weapon/gun/energy/plasmacutter))
-		sawoff(user)
+		temp_sawing = 1
 	if(istype(A, /obj/item/weapon/melee/energy))
 		var/obj/item/weapon/melee/energy/W = A
 		if(W.active)
-			sawoff(user)
+			temp_sawing = 1
+	if(temp_sawing)
+		if(style == 2)

```

Comment:
There are only two styles. Make this more clear by changing it to `var/sawedoff` or something similar and just using boolean values. `if(!sawedoff) sawoff`

Reply:
Style is now just used inside the proc as a quick way to say what the tool does, such as saws using 0, and pipes being 1